panic on updateMsgAppTerm because streamReader has not been inited

=== RUN TestDoubleClusterSizeOf3-4
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x5eb382]
goroutine 128691 [running]:
github.com/coreos/etcd/rafthttp.(*streamReader).updateMsgAppTerm(0x0, 0x2)
	/home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/stream.go:375 +0x102
github.com/coreos/etcd/rafthttp.(*peer).setTerm(0xc21430bf80, 0x2)
	/home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/peer.go:210 +0x36
github.com/coreos/etcd/rafthttp.(*transport).maybeUpdatePeersTerm(0xc21430ba80, 0x2)
	/home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/transport.go:124 +0xca
github.com/coreos/etcd/rafthttp.(*transport).Send(0xc21430ba80, 0xc20aaa5500, 0x1, 0x1)
	/home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/rafthttp/transport.go:137 +0x15a
github.com/coreos/etcd/etcdserver.(*EtcdServer).send(0xc20edf17a0, 0xc20aaa5500, 0x1, 0x1)
	/home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:725 +0x102
github.com/coreos/etcd/etcdserver.(*raftNode).run(0xc20edf17b0)
	/home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/raft.go:161 +0x85f
created by github.com/coreos/etcd/etcdserver.(*EtcdServer).run
	/home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:419 +0x241
goroutine 1 [chan receive]:
testing.RunTests(0xa47f28, 0xbd5e60, 0x22, 0x22, 0x1)
	/usr/local/go/src/testing/testing.go:556 +0xad6
testing.(*M).Run(0xc208056690, 0xbe09e0)
	/usr/local/go/src/testing/testing.go:485 +0x6c
main.main()
	github.com/coreos/etcd/integration/_test/_testmain.go:118 +0x1d5

--- a/rafthttp/peer.go
+++ b/rafthttp/peer.go
@@ -143,10 +143,10 @@ func startPeer(tr http.RoundTripper, urls types.URLs, local, to, cid types.ID, r
 		}
 	}()
 
+	p.msgAppReader = startStreamReader(tr, picker, streamTypeMsgAppV2, local, to, cid, status, p.recvc, p.propc, errorc)
+	reader := startStreamReader(tr, picker, streamTypeMessage, local, to, cid, status, p.recvc, p.propc, errorc)
 	go func() {
 		var paused bool
-		p.msgAppReader = startStreamReader(tr, picker, streamTypeMsgAppV2, local, to, cid, status, p.recvc, p.propc, errorc)
-		reader := startStreamReader(tr, picker, streamTypeMessage, local, to, cid, status, p.recvc, p.propc, errorc)
 		for {
 			select {
 			case m := <-p.sendc:

